app/src/main/java/com/example/lapnguyen/weather_3/api/GPSTracker.java

```diff
@@ -0,0 +1,161 @@
+package com.example.lapnguyen.weather_3.api;
+
+/**
+ * Created by lapnguyen on 16/06/2017.
+ */
+
+import android.Manifest;
+import android.app.AlertDialog;
+import android.app.Service;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.provider.Settings;
+import android.support.v4.app.ActivityCompat;
+import android.util.Log;
+
+public class GPSTracker extends Service implements LocationListener {
+
+    private final Context mContext;
+
+    // flag for GPS status
+    boolean isGPSEnabled = false;
+
+    // flag for network status
+    boolean isNetworkEnabled = false;
+
+    // flag for GPS status
+    boolean canGetLocation = false;
```

Comment:
WARNING: [AvoidFieldNameMatchingMethodName] Field canGetLocation has the same name as a method
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName